internal/controllers/framework_controller_test.go

```diff
@@ -76,7 +76,8 @@ func TestFrameworkReconciler_Reconcile(t *testing.T) {
 			},
 			wantStatusPhase: ketchv1.FrameworkCreated,
 			wantNamespaceLabels: map[string]string{
-				"istio-injection": "enabled",
+				"istio-injection":             "enabled",
+				"kubernetes.io/metadata.name": "another-namespace-3",
```

Comment:
I've got this running it locally
```bash
--- FAIL: TestFrameworkReconciler_Reconcile (13.60s)
    --- FAIL: TestFrameworkReconciler_Reconcile/istio_controller_-_everything_is_ok (1.26s)
        framework_controller_test.go:149: 
            	Error Trace:	framework_controller_test.go:149
            	Error:      	Not equal: 
            	            	expected: map[string]string{"istio-injection":"enabled", "kubernetes.io/metadata.name":"another-namespace-3"}
            	            	actual  : map[string]string{"istio-injection":"enabled"}
            	            	
```
it makes me thinking it depends on kubebuilder version. 
Does it make sense to remove "kubernetes.io/metadata.name" from `gotNamespace` before comparing its labels with `wantNamespaceLabels`
https://github.com/theketchio/ketch/blob/1a5c8f19c255ae1bb7b18d0e5c198dfac6509609/internal/controllers/framework_controller_test.go#L149

Reply:
its because we are using a very outdated version of `kubebuilder` I believe which has outdated versions of `kube-apiserver` and `etcd` bundled as binaries in the bin directory. I feel that moving to the latest version would be the right move here

Reply:
it was kustomize. I udpdated my local version to 4.5 and it works now
should we change the version of `kube-apisever` there
https://github.com/theketchio/ketch/blob/4b313a63ff9efddec0f93820c02816d4f2da9b7f/internal/controllers/suite_test.go#L58

Reply:
yep bumped this up to 1.24.0

Reply:
yep bumped this up to 1.24.0